### DISCUSSION TITLE
in ts getters using 'this' doesn't affect

### DISCUSSION THREAD

**Discussion by squarexyz** [vuejs/pinia#2538] [Help and Questions] [answered]
### Reproduction

https://play.pinia.vuejs.org/#eNqNVE1vGjEQ/SvWXoAG7KS0UYUgoq1ySA9t1fS4l/0YwMQeW7aXIKH97521YSFpFfW2nnnz5s2zZw/ZZ2v5roFsls195aQNzENoLFMFrhd5Fnye3eUotTUusANrPDwG44C1bOWMZgMuKtNgAMeDH+SYY2XQE0kELXr8cJTjXKQOxEeHANqqIgCdGJuXTQgG2bJSsnqivrGe49UVtX/AyoEGpP6HRMyRte1cpCIimIueLRtnSexEF5ZvvUEa7dD1yI8JmmjGYqSLWYmy6CJ5JuI3B68npTPPnmbaEnh8gi7JJ1HDLhij/KSwMpVtQrB+JkRVI+FrUHLnOEIQaLX4q2Z5yz/yG6FkKaiRkFjD/mUbqpjUoP+H/QRdXvObD/w2su6mR1LdsXakbY4t2RI8Xc1Krl+ZUhltpQL3wwZJV/fCnEIp8/wtxoJroJdYbaB6+kd86/dJ9U8HZN8OLsYKhVtDSOn7x++wp+8+qU3dKEK/kfwF3qim05hgXxqsSfYFLqp9iHcscf3b3+8DoD8N1QmNbkR8tPnrG6Of5U759MLF82snC/u1qGEl8dVmxNdEK9GDHKz6JHWP2wL7mExL0y/X4pJwODj2HIyTPB/ooc/YcMQWd2x4oGJKz9g1a0dxNtJNaN8PU5umVF1BLBzNGDa6BNdVxwiPBOwde3+0siirjv4SGDbS80REwE9nI4sqetc3k6dlJXnHEEvVsQstdIyRm8lS+i9k7R8SN4D2

### Steps to reproduce the bug

<img width="785" alt="image" src="https://github.com/vuejs/pinia/assets/134040412/5315e32b-59fb-440b-9335-e101ee110e07">

"pinia": "^2.1.7",
"node": "21.5.0"
 "typescript": "~5.3.0",
  "vite": "^5.0.10",
  "vitest": "^1.0.4",
  "vue-tsc": "^1.8.26"

// tsconfig
"compilerOptions": {
    "composite": true,
    "noEmit": true,
    "baseUrl": ".",
    "strict": true,
    "paths": {
      "@/*": ["./src/*"]
    }
  }

### Expected behavior

use this autocompletion and typings for the whole store 

### Actual behavior

this not affect

**Comment by posva** [comment, marked as answer]
`this` in arrow functions is different: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

[You just need to use a regular function](https://play.pinia.vuejs.org/#eNqNVE1v2zAM/SuCL0nWRGqXrRiMpMg29NAdtmHd0Rd/MIlSixIkOQ0Q+L+Plhwn7YZiN5t8fHx8EnVMPhvD9w0kabJwpZXGMwe+MazOcbPMEu+y5C5DqYy2nh1Z4+DRawusZWurFRtxUeoGPVju3SjDDEuNjkgCaDngx5MMFyJ2ID768aBMnXugP8YWReO9RrYqa1k+Ud9Qz/Hqito/YGlBAVL/YyTmyNp2IWIRESzEwJZMkyh2pnLDd04jjXbsemR9giZKWYh0MSNR5l0kS0T45uDUrLD62dFMOwJPT9AV+SQq2HutazfLjYxlW++NS4UoKyR8BbXcW47gBRol/qpZ3fKP/EbUshDUSEis4PCyDVXMKlD/w36Crq75zQd+G1j3855UdawdaZthS7Z4R0ezlptXppRaGVmD/WG8pKN7YU5e1/r5W4h528AgsdxC+fSP+M4douqfFsi+PVyM5XO7AR/T94/f4UDfQ1LpqqkJ/UbyFzhdN53GCPvSYEWyL3BB7UM4Y4mb3+7+4AHdaahOaHAj4IPNX98Y/Sx3zucXLp5vO1k4rEUFa4mvNiPcJlqJAWRhPSSpe9gWOIRkXJphuZaXhONR33M0jfKcp4uesvGELe/Y+EjFlE7ZNWsnYTbSTWg3DFPppqi7glA4SRk2qgDbVYcIDwTsHXvfW5kX5fgM61ksPQsWmd9KxyMjVXyKOfLmZGxeBi+H5vK0vCS3D7FIErrSgl8wkMX0TiTtH3+uhk0=)
